src/StorageType.cs

```diff
@@ -4,11 +4,11 @@
 
 namespace HypEcs;
 
-public struct StorageType : IComparable<StorageType>
+public readonly struct StorageType : IComparable<StorageType>
 {
-    public Type Type { get; private set; }
-    public ulong Value { get; private set; }
-    public bool IsRelation { get; private set; }
+    public readonly Type Type;
```

Comment:
Suggest easier change:
```cs
    public required Type Type { get; init; }
    public required ulong Value { get; init; }
    public required bool IsRelation { get; init; }
```

(done! all other changes unnecessary)

Reply:
This wouldn't compile in unity since it only has C# 9.0 support and this is a C# 11.0 feature.